src/applications/claims-status/tests/components/DocumentRequestPage.unit.spec.jsx

```diff
@@ -29,296 +29,391 @@ const defaultProps = {
-      body: 'test',
-      type: 'error',
-    };
-    const clearNotification = sinon.spy();
-
-    const tree = SkinDeep.shallowRender(
-      <DocumentRequestPage
-        {...defaultProps}
-        trackedItem={trackedItem}
-        uploadComplete
-        clearNotification={clearNotification}
-        message={message}
-      />,
-    );
-    expect(tree.subTree('Notification')).not.to.be.false;
-    tree.getMountedInstance().componentWillUnmount();
-    expect(clearNotification.called).to.be.false;
-  });
+    it('should render loading div', () => {
+      const { context } = renderWithRouter(
+        <Provider store={getStore(false)}>
+          <DocumentRequestPage {...defaultProps} loading />,
+        </Provider>,
+      );
 
-  it('should render due date info', () => {
-    const trackedItem = {
-      status: 'NEEDED_FROM_YOU',
-      suspenseDate: '2010-05-10',
-    };
+      expect($('va-loading-indicator', context)).to.exist;
+      expect($('.claim-container', context)).to.not.exist;
+    });
 
-    const tree = SkinDeep.shallowRender(
-      <DocumentRequestPage {...defaultProps} trackedItem={trackedItem} />,
-    );
+    it('should render upload error alert', () => {
+      const trackedItem = {
+        status: 'NEEDED_FROM_YOU',
+      };
+      const message = {
+        title: 'Test',
+        body: 'Testing',
+      };
+
+      const { context } = renderWithRouter(
+        <Provider store={getStore(false)}>
+          <DocumentRequestPage
+            {...defaultProps}
+            trackedItem={trackedItem}
+            message={message}
+          />
+          ,
+        </Provider>,
+      );
+      expect($('va-alert', context)).to.exist;
+    });
 
-    expect(tree.subTree('DueDate')).not.to.be.false;
-    expect(tree.subTree('DueDate').props.date).to.eql(trackedItem.suspenseDate);
-  });
+    it('should render upload error alert when rerendered', () => {
+      const trackedItem = {
+        status: 'NEEDED_FROM_YOU',
+      };
+
+      const { container, rerender } = renderWithRouter(
+        <Provider store={getStore(false)}>
+          <DocumentRequestPage {...defaultProps} trackedItem={trackedItem} />,
+        </Provider>,
+      );
+      expect($('va-alert', container)).not.to.exist;
+
+      const message = {
+        title: 'Test',
+        body: 'Testing',
+      };
+
+      rerenderWithRouter(
+        rerender,
+        <Provider store={getStore(false)}>
+          <DocumentRequestPage
+            {...defaultProps}
+            trackedItem={trackedItem}
+            message={message}
+          />
+          ,
+        </Provider>,
+      );
+      expect($('va-alert', container)).to.exist;
+      expect($('va-alert h2', container).textContent).to.equal(message.title);
+    });
 
-  it('should render optional upload alert', () => {
-    const trackedItem = {
-      status: 'NEEDED_FROM_OTHERS',
-      suspenseDate: '2010-05-10',
-    };
+    // it('should clear upload error when leaving', () => {
```

Comment:
@jerekshoe had trouble getting this test to refactor and when I did it was passing when I ran it alone and then failing when I ran it with the other tests. let me know your thoughts!